### PR TITLE
Update turbulent taylor couette time stepping in example

### PR DESCRIPTION
**Pull request by blaisb**
### Description

The Taylor-Couette example was using constant time-stepping and a BDF3 time-integrator. This is not a good practice for this example. I have updated the parameter file, the prm file and also the .png files of the results. The new results are much better especially for fine meshes.

Code related list:
- [X] Lethe documentation is up to date
- [X] The branch is rebased onto master
- [X] Code is indented with indent-all and .prm files (examples and tests) with prm-indent
- [X] Links are added to parent .rst files
- [X] The example is following the [standard format](https://chaos-polymtl.github.io/lethe/documentation/contributing.html#general-rules-and-format)

Pull request related list:
- [X] Labels are applied
- [X] There are at least 2 reviewers (or 1 if small feature) excluding the responsible for the merge
- [X] If this PR closes an issue or is related to a project, it is linked in the "Projects" or "Development" section
- [X] If any future works is planed, an issue is opened
- [X] The PR description is cleaned and ready for merge

### CODE REVIEW

File: examples/incompressible-flow/3d-turbulent-taylor-couette/tc-matrix-free.prm [hunk 1]
```diff
@@ -139,9 +148,9 @@ subsection linear solver
     set verbosity         = verbose
 
     # MG parameters
-    set mg verbosity       = quiet
-    set mg min level       = -1
-    set mg level min cells = 16
+    set mg verbosity                   = quiet
+    set mg min level                   = -1
+    set mg level min cells             = 16
```
Comment (lpsaavedra) [inline examples/incompressible-flow/3d-turbulent-taylor-couette/tc-matrix-free.prm:154]: This indentation seems odd to me. Are you sure that you used the prmindent file?

File: doc/source/examples/incompressible-flow/3d-turbulent-taylor-couette/images/enstrophy_comparison_655k_Q2.png
[the remark this answers is not in the document]
Comment (blaisb) [reply doc/source/examples/incompressible-flow/3d-turbulent-taylor-couette/images/enstrophy_comparison_655k_Q2.png:1]: Makes sense, I will fix that

File: examples/incompressible-flow/3d-turbulent-taylor-couette/tc-matrix-free.prm (hunk 1 above)
Comment (blaisb) [reply examples/incompressible-flow/3d-turbulent-taylor-couette/tc-matrix-free.prm:154]: Makes sense, I will fix that

**Comment by blaisb** [comment]
I applied all of your comments @lpsaavedra

**Review by lpsaavedra — APPROVED** [review]

**Pull request merged**